core/domain/wipeout_service.py

```diff
@@ -339,18 +345,163 @@ def _pseudonymize_models(activity_related_models, pseudonymized_user_id):
+    ).fetch()
+    feedback_ids |= set([model.thread_id for model in feedback_message_models])
+
+    suggestion_model_class = suggestion_models.GeneralSuggestionModel
+    general_suggestion_models = suggestion_model_class.query(ndb.OR(
+        suggestion_model_class.author_id == user_id,
+        suggestion_model_class.final_reviewer_id == user_id
+    )).fetch()
+    feedback_ids |= set([model.id for model in general_suggestion_models])
+
+    # The activity_mappings field might have only been partially generated, so
+    # we fill in the missing part for this activity category.
+    if models.NAMES.feedback not in pending_deletion_request.activity_mappings:
+        pending_deletion_request.activity_mappings[models.NAMES.feedback] = (
+            _generate_activity_to_pseudonymized_ids_mapping(feedback_ids)
+        )
+        save_pending_deletion_request(pending_deletion_request)
+
+    def _pseudonymize_models(feedback_related_models, pseudonymized_user_id):
+        """Pseudonymize user ID fields in the models.
+
+        Args:
+            feedback_related_models: list(BaseModel). Models whose user IDs
+                should be pseudonymized.
+            pseudonymized_user_id: str. New pseudonymized user ID to be used for
+                the models.
+        """
+        feedback_thread_models = [
+            model for model in feedback_related_models
+            if isinstance(model, feedback_thread_model_class)]
+        for feedback_thread_model in feedback_thread_models:
+            if feedback_thread_model.original_author_id == user_id:
+                feedback_thread_model.original_author_id = pseudonymized_user_id
+            if feedback_thread_model.last_nonempty_message_author_id == user_id:
+                feedback_thread_model.last_nonempty_message_author_id = (
+                    pseudonymized_user_id)
+
+        feedback_message_models = [
+            model for model in feedback_related_models
+            if isinstance(model, feedback_message_model_class)]
+        for feedback_message_model in feedback_message_models:
+            feedback_message_model.author_id = pseudonymized_user_id
+
+        general_suggestion_models = [
+            model for model in feedback_related_models
+            if isinstance(model, suggestion_model_class)]
+        for general_suggestion_model in general_suggestion_models:
+            if general_suggestion_model.author_id == user_id:
+                general_suggestion_model.author_id = pseudonymized_user_id
+            if general_suggestion_model.final_reviewer_id == user_id:
+                general_suggestion_model.final_reviewer_id = (
+                    pseudonymized_user_id)
+
+        ndb.put_multi(
+            feedback_thread_models +
+            feedback_message_models +
+            general_suggestion_models
+        )
+
+    feedback_mappings = (
+        pending_deletion_request.activity_mappings[models.NAMES.feedback])
```

Comment:
I am finding the name activity_mappings (and feedback_mappings, for that matter) hard to understand, as a reader of the code. I have no idea what their internal structure is. Could you replace them with something more specific that explains what is actually in the dict -- e.g., for the latter, something like feedback_ids_to_pseudonymized_user_ids or feedback_ids_to_pids, perhaps?